docs/changes/16181.other.rst

```diff
@@ -0,0 +1,30 @@
+Public APIs that expose a ``copy`` argument and that previously set ``False``
+as a default value now use ``None`` instead if Numpy 2.0 or newer is installed.
+This is because in Numpy 2.0, the meaning of the ``copy`` argument has changed,
+with ``copy=False`` now indicating that a copy should never be made, while
+``copy=None`` is used for the previous meaning of "avoid a copy if possible".
+
+This includes:
+- ``astropy.units.Quantity``
+- ``astropy.utils.Masked``
+- ``astropy.table.Column``
+- ``astropy.time.Time``
+- ``astropy.coordinates.SkyCoord``
+
+While this change ensures the default behaviour of astropy has not changed,
+code that explicitly passes ``copy=False`` to many of astropy's classes
+will need to be adjusted where the intention was to forbid unnecessary copies
+but allow the ones that couldn't be avoided.
+For portability across different versions of Numpy, we recommend that these
+instances of ``False`` be replaced with a ``COPY_IF_NEEDED`` constant defined
+as follow
+
+    def NUMPY_GE_2_0() -> bool:
```

Comment:
Can we just recommend a simpler version?
```
COPY_IF_NEEDED = False if np.__version__[0] == "1" else None
```

Reply:
Given how these compatibility layer tend to be left to rot in unmaintained software, I'm slightly uneasy to recommend something that would break in the event of numpy going to 3.0. I'm also hesitant to recommend more than one way to do it, which is why I went with the more robust alternative.

Reply:
hum, on second thought I think this would only break if there was a Numpy 10.0 ... and by then, anything that still has compatibility code for Numpy 1.x is likely to be broken anyway so, let's just roll with it !

Reply:
and on **third** thought, we can make it even more portable at no real addtional cost:
```python
COPY_IF_NEEDED = False if np.__version__.startswith("1.") else None
```